auction9-backend/handler.js

```diff
@@ -171,15 +171,76 @@ export const stopActiveAuction = async (event, context) => {
  * GET: /wonauctions
  */
 export const getUserWonAuctions = async (event, context) => {
-   try {
-     let currentUserId = event.multiValueQueryStringParameters.userId[0];
-     let resultsUserWonAuctions = await mysql.query(`SELECT * FROM tbl_auction WHERE winner=?`, [currentUserId]);
-     await mysql.end();
-     return generateResponse(200, resultsUserWonAuctions);
-   } catch (error) {
-     console.log(error);
-     return generateResponse(400, {
-       message: 'There was an error while getting my won auctions.'
-     });
-   }
+  try {
+    let currentUserId = event.multiValueQueryStringParameters.userId[0];
+    let resultsUserWonAuctions = await mysql.query(`SELECT * FROM tbl_auction WHERE winner=?`, [currentUserId]);
+    await mysql.end();
+    return generateResponse(200, resultsUserWonAuctions);
+  } catch (error) {
+    console.log(error);
+    return generateResponse(400, {
+      message: 'There was an error while getting my won auctions.'
+    });
+  }
+};
+
+
+/* Allowed status changes:
+ * order status: INACTIVE -> ACTIVE -> FINISHED -> REALIZED
+ * current -> new :
+ * INACTIVE -> ACTIVE   [x]
+ * ACTIVE -> INACTIVE   [x]
+ * ACTIVE -> FINISHED   [x]
+ * FINISHED -> REALIZED [x]
+ */
+/* realizeFinishedAuction - will update status for auction to 'realized'
+ * PUT: /myauctions/id
+ */
+export const realizeFinishedAuction = async (event, context) => {
+  try {
+    let reqBody = JSON.parse(event.body);
+    let auctionId = reqBody.auction.auctionID;
+    let currentStatus = reqBody.auction.status;
+    // requested status to be changed into
+    let reqStatus = reqBody.changeStatus;
+    let possibleStatus = ['INACTIVE', 'ACTIVE', 'FINISHED', 'REALIZED'];
```

Comment:
The possible statuses can be an object, it will be easier when you want to point out some of them.

example:
statuses: {
'inactive': 'INACTIVE',
'active': 'ACTIVE,
...
}

then you can use in if statement like this: `if (reqStatus === statuses.INACTIVE)`

Reply:
Changed to be an object